packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -653,10 +703,36 @@ class FuchsiaDevice extends Device {
           ports.add(port);
         }
       }
+
+      // Get component urls
```

Comment:
You might find tests easier to write if you pull some of this logic out into separate helper methods. They can be annotated `@visibleForTesting` to make them visible only to tests and other code in this Dart library.

Reply:
Agreed, I will refactor this.